product_analytic_purchase/models/purchase_order_line.py

```diff
@@ -7,22 +7,32 @@
 class PurchaseOrderLine(models.Model):
     _inherit = "purchase.order.line"
 
-    @api.onchange("product_id")
-    def onchange_product_id(self):
-        res = super(PurchaseOrderLine, self).onchange_product_id()
-        if self.product_id:
-            ana_accounts = (
-                self.product_id.product_tmpl_id._get_product_analytic_accounts()
-            )
-            ana_account = ana_accounts["expense"]
-            self.account_analytic_id = ana_account.id
-        return res
+    def _compute_analytic_distribution(self):
+        """
+        Get analytic distribution from product expense analytic account
+        If no account is set, call super with those records
+        """
+        lines_without_analytic_ids = set()
+        for line in self:
+            if line.product_id:
+                ana_accounts = (
+                    line.product_id.product_tmpl_id._get_product_analytic_accounts()
+                )
+                ana_account = ana_accounts["expense"]
+                if ana_account:
+                    line.analytic_distribution = {ana_account.id: 100}
+                else:
+                    lines_without_analytic_ids.add(line.id)
+        return super(
+            PurchaseOrderLine, self.browse(lines_without_analytic_ids)
+        )._compute_analytic_distribution()
 
-    @api.model
-    def create(self, vals):
-        if vals.get("product_id") and not vals.get("account_analytic_id"):
-            product = self.env["product.product"].browse(vals.get("product_id"))
-            ana_accounts = product.product_tmpl_id._get_product_analytic_accounts()
-            ana_account = ana_accounts["expense"]
-            vals["account_analytic_id"] = ana_account.id
-        return super(PurchaseOrderLine, self).create(vals)
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get("product_id") and not vals.get("analytic_distribution"):
+                product = self.env["product.product"].browse(vals.get("product_id"))
+                ana_accounts = product.product_tmpl_id._get_product_analytic_accounts()
+                ana_account = ana_accounts["expense"]
+                vals["analytic_distribution"] = {ana_account.id: 100.0}
```

Comment:
This would crash if ana_account is False I guess we should check the value before accessing `.id`.

Reply:
@IT-Ideas Done